frontend/src/App.js

```diff
@@ -1,38 +1,62 @@
-import React, { Component } from "react";
+import React, { Component, useEffect, useState } from "react";
 import "./App.css";
 import CustomTable from "./CustomTable.js";
 import PropTypes from "prop-types";
 import EmptyRoomsTable from "./EmptyRoomsTable.js";
 import TwoSlotDisplay from "./TwoSlotDisplay.js";
 import Logo from "./navbar-icon.svg";
+import { isInsideCampusNetwork } from "./Utilities.js";
 
-class App extends Component {
-  static propTypes = {
-    schedule: PropTypes.object.isRequired,
-    empty_schedule: PropTypes.array.isRequired
-  };
+function App({ schedule, empty_schedule }) {
+  const [loading, setLoading] = useState(true);
+  const [show, setShow] = useState(false);
 
-  render() {
-    return (
-      <div className="App">
-        <header className="App-header">
-          <img src={Logo} alt="logo"></img>
-          <h1>Chillzone - IIT Kharagpur</h1>
-          <h3>Find a place to chill, NOW!</h3>
-        </header>
-
-        <TwoSlotDisplay
-          date={new Date()}
-          schedule={this.props.empty_schedule}
-        />
+  useEffect(() => {
+    isInsideCampusNetwork()
+      .then((value) => {
+        setShow(value);
+        setLoading(false);
+      })
+      .catch(() => setLoading(false));
+  }, []);
 
-        <EmptyRoomsTable schedule={this.props.empty_schedule} />
-        {Object.keys(this.props.schedule).map(key => {
-          return <CustomTable room={key} schedule={this.props.schedule[key]} />;
-        })}
+  if (loading)
+    return (
+      <div className="message-banner">
+        <iframe src="https://lottie.host/?file=ca9a3787-e6db-4878-a0d3-d10dde95b225/MjnHw4B5gw.json"></iframe>
```

Comment:
Is it safe to load an iframe (is it vulnerable to man in the middle)? Isn't it possible to embed this image directly?

Reply:
[Not really](https://stackoverflow.com/questions/7289139/why-are-iframes-considered-dangerous-and-a-security-risk). 
Yes (problem is, it's not an image. will try to download gif and use them).

Reply:
> it's not an image
:hmm: